packages/suite/src/support/suite/ThemeProvider.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from 'react';
+import { ThemeProvider as SCThemeProvider } from 'styled-components';
+import { THEME } from '@trezor/components';
+import * as suiteActions from '@suite-actions/suiteActions';
+import { useSelector, useActions } from '@suite-hooks';
+import { getOSTheme } from '@suite-utils/dom';
+import { db } from '@suite/storage';
+import { AppState } from '@suite-types';
+
+const getThemeColors = (theme: AppState['suite']['settings']['theme']) => {
+    switch (theme?.variant) {
+        case 'light':
+            return THEME.light;
+        case 'dark':
+            return THEME.dark;
+        case 'custom':
+            // custom theme is a secret feature accessible in debug settings
+            return {
+                ...THEME.dark, // spread default colors, so we can be sure no new colors are missing in user's palette
+                ...theme.colors, // custom saved colors
+            };
+        default:
+            return THEME.light;
```

Comment:
I am thinking about whether it won't be better to do migration and fill store with theme default. But this is probably good enough as long as somebody does not remove this default.

Reply:
yeah, if we are gonna do some changes we are supposed to do proper migration. this is just for the events when we forgot about it :)

Reply:
ohh wait. you werent talking about the `default` case. give me a sec.

Reply:
that spread in `custom` case is really a safety check, as the feature is intended only for internal use I wanted to prevent weird things happening to our beloved product team. I do not expect we are going to bother with migrations for this hidden top secret interal-only feature.